feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -743,14 +745,14 @@ class BillingWrapper(
         )
 
         synchronized(this@BillingWrapper) {
-            val presentedOffering = presentedOfferingsByProductIdentifier[purchase.firstProductId]
-            val subscriptionOptionId = subscriptionOptionSelectedByProductIdentifier[purchase.firstProductId]
-            productTypes[purchase.firstProductId]?.let { productType ->
+            val context = purchaseContext[purchase.firstProductId]
+            context?.productType?.let { productType ->
                 completion(
                     purchase.toStoreTransaction(
-                        productType,
-                        presentedOffering,
-                        subscriptionOptionId
+                        context.productType,
+                        context.presentedOffering,
+                        context.subscriptionOptionSelected,
+                        context.prorationMode,
```

Comment:
Wonder if we could just make this `toStoreTransaction` receive a `PurchaseContext` object?

Reply:
I added an overload with it, but there are a lot of use cases that don't take a PurchaseContext, or mix from different sources, so I'd still leave the other creator.